src/gei/Commands/MigrateRepo/MigrateRepoCommandHandler.cs

```diff
@@ -221,7 +221,7 @@ private string ExtractGhesBaseUrl(string ghesApiUrl)
         var (gitArchiveUrl, metadataArchiveUrl, gitArchiveId, metadataArchiveId) = await _retryPolicy.Retry(
             async () => await GenerateArchive(githubSourceOrg, sourceRepo, skipReleases, lockSourceRepo));
 
-        if (!blobCredentialsRequired)
+        if (!useGithubStorage && !blobCredentialsRequired)
```

Comment:
We need a unit test for a scenario where `useGithubStorage` is `true` and `blobCredentialsRequired` is `false`